crates/shared/src/http_solver_api.rs

```diff
@@ -67,10 +67,10 @@ impl HttpSolverApi for DefaultHttpSolverApi {
         let timeout = deadline
             .checked_duration_since(Instant::now())
             .ok_or_else(|| anyhow!("no time left to send request"))?;
-        // The timeout we give to the solver is one second less than the deadline to make up for
-        // overhead from the network.
+        // The timeout we give to the solver is a few milliseconds less than
+        // the deadline to make up for overhead from the network.
         let solver_timeout = timeout
-            .checked_sub(Duration::from_secs(1))
+            .checked_sub(Duration::from_millis(50))
```

Comment:
Is 50 ms enough? Maybe 100ms might be a bit safer. Either way, I'm OK with lowering it from 1s which seemed a bit high.